orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/orca/q/redis/migration/TaskTypeDeserializer.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.q.redis.migration
+
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.JsonDeserializer
+import com.fasterxml.jackson.databind.exc.InvalidFormatException
+import com.netflix.spinnaker.orca.Task
+import com.netflix.spinnaker.orca.TaskResolver
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+
+class TaskTypeDeserializer(
```

Comment:
I don't yet know enough about deserializers to know if this is sufficient scoped (vs say implementing a deserializer for the entirety of a `RunTask` message)

Reply:
This should work just fine - I like it at this granularity.